examples/src/test/kotlin/UpdatesBuildersTest.kt

```diff
@@ -0,0 +1,259 @@
+        // :snippet-end:
+        assertEquals(8, getDocument().qty)
+    }
+
+    @Test
+    fun mulUpdateTest() = runBlocking {
+         // :snippet-start: mul-update
+        val filter = Filters.eq("_id", 1)
+        val update = Updates.mul(PaintOrder::qty.name, 2)
+        collection.updateOne(filter, update)
+        // :snippet-end:
+        assertEquals(10, getDocument().qty)
+    }
+
+    @Test
+    fun renameUpdateTest() = runBlocking {
+        // :snippet-start: rename-update
+        val filter = Filters.eq("_id", 1)
+        val update = Updates.rename(PaintOrder::qty.name, "quantity")
+        collection.updateOne(filter, update)
+        // :snippet-end:
+        assertEquals(null, getDocument().qty)
+        val doc = collection.find<Document>(filter).first()
+        assertEquals(5, doc.getInteger("quantity"))
+
+    }
+
+    @Test
+    fun minUpdateTest() = runBlocking {
+        // :snippet-start: min-update
+        val filter = Filters.eq("_id", 1)
+        val update = Updates.min(PaintOrder::qty.name, 2)
+        collection.updateOne(filter, update)
+        // :snippet-end:
+        assertEquals(2, getDocument().qty)
+    }
+
+    @Test
+    fun maxUpdateTest() = runBlocking {
+        // :snippet-start: max-update
+        val filter = Filters.eq("_id", 1)
+        val update = Updates.max(PaintOrder::qty.name, 8)
+        collection.updateOne(filter, update)
+        // :snippet-end:
+        assertEquals(8, getDocument().qty)
+    }
+
+    @Test
+    fun currentDateUpdateTest() = runBlocking {
+        // :snippet-start: current-date-update
+        val filter = Filters.eq("_id", 1)
+        val update = Updates.currentDate(PaintOrder::lastModified.name)
+        collection.updateOne(filter, update)
+        // :snippet-end:
+        val docTime = getDocument().lastModified!!.time
+        val nowTime = Date().time
+        val diff = nowTime - docTime
+        assert(diff < 10000)
+
+    }
+
+    @Test
+    fun currentTimestampUpdateTest() = runBlocking {
+        // :snippet-start: current-timestamp-update
+        // Create a new instance of the collection with the flexible `Document` type
+        // to allow for the changing of the `lastModified` field to a `BsonTimestamp`
+        // from a `Date`.
+        val collection = database.getCollection<Document>("paint_orders")
+
+        val filter = Filters.eq("_id", 1)
+        val update = Updates.currentTimestamp(PaintOrder::lastModified.name)
+        collection.updateOne(filter, update)
+        // :snippet-end:
+        val docTime = collection.find().first().get("lastModified", BsonTimestamp::class.java).time
+        val nowTime = BsonTimestamp().time
+        val diff = nowTime - docTime
+        assert(diff < 10000)
+    }
+
+    @Test
+    fun bitwiseOrUpdateTest() = runBlocking {
+        // :snippet-start: bitwise-or-update
+        val filter = Filters.eq("_id", 1)
+        val update = Updates.bitwiseOr(PaintOrder::qty.name, 10)
```

Comment:
You wouldn't use bitwise operators on a quantity.  Do we care though?

Reply:
i would say, yes we should care. but no we should not change it right now b/c this is just copying the [java example](https://www.mongodb.com/docs/drivers/java/sync/current/fundamentals/builders/updates/#bit).

what would a more sensible example be here? i'll make a follow up ticket to update it